src/components/note/__internal__/status-with-tooltip/status.component.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import StyledStatusIconWrapper from './status.style';
+import TooltipDecorator from '../../../../utils/decorators/tooltip-decorator/tooltip-decorator';
+
+class StatusIcon extends React.Component {
```

Comment:
Why is this a class and no a functional component? I'm very against creating new class components unless it's absolutely necessary

Reply:
Yeah completely agree, but the tooltip decorator requires a class component so it can hook into `this`

Reply:
Ah fair enough